### ISSUE TITLE
compiled code

### ISSUE THREAD

**Issue by zeffii** [Sverchok/SverchokRedux#28]
https://docs.python.org/3/extending/extending.html

shows how to compile C code that uses `<Python.h>` to make a module for importing in Python.
worth thinking about this kind of stuff for heavier ops.

Here I transcribed two youtube videos which very lightly explain the topic.  https://github.com/zeffii/Learning_C_TheHardWay_studynotes/blob/master/python_cmodule.md

http://stackoverflow.com/questions/25459479/how-to-return-an-c-array-to-python

**Comment by zeffii** [comment]
if you where to make a C function it could also just return a flat object

The first float would mean '3 slices`
The next three floats would then 'broadcast' how many vertices per each slice
The subsequent floats would be 1 D array of coordinate components.
[3.0, 9.0, 8.0, 9.0, -1.0, -1.0, -1.0 ........... 1.0, 1.0, 1.0] 

![image](https://cloud.githubusercontent.com/assets/619340/12394888/dd07d7a4-bdfe-11e5-9763-1bb597121a21.png)

**Comment by zeffii** [comment]
![image](https://cloud.githubusercontent.com/assets/619340/12395081/eeb437bc-bdff-11e5-84ab-6ddeafc0d8fc.png)

**Comment by ly29** [comment]
Furthermore it could used to access & provide numpy objects.
http://docs.scipy.org/doc/numpy-1.10.0/reference/c-api.array.html

**Comment by zeffii** [comment]
yeah! can probably return pyObjects that are multiple datatypes, a `SvRxCompiled.make_cube` could return `geom` .. just thinking out loud. (not sure this is possible...or easy... or the best approach)

**Comment by zeffii** [comment]
https://docs.python.org/3/extending/extending.html#building-arbitrary-values
-- from that doc it doesn't look very convenient to return arbitrary lists.. because it needs the formatting string and the content in the `Py_BuildValue("[i,i]", 123, 456)` . so yeah --- numpy c api might be the option.
